src/brod_group_coordinator.erl

```diff
@@ -1007,12 +1009,16 @@ make_offset_commit_metadata() ->
 %% then received by group leader, group leader (may mutate it and)
 %% assigns it back to members in topic-partition assignments.
 %%
-%% Currently user_data is created in this module and terminated in this
-%% module, when needed for advanced features, we can originate it from
-%% member's init callback, and pass it to members via
-%% `brod_received_assignments()'
--spec user_data(join | assign) -> binary().
-user_data(_Action) -> <<"">>.
+%% Currently user_data in join requests can be created from a
+%% callback to member module.
+%% user_data assigned back to members is always an empty binary
+%% if assign_partitions callback is not implemented, otherwise
+%% should be set in the assign_partitions callback implemenation.
+user_data(Module, Pid) ->
+  case lists:member({user_data, 1}, Module:module_info(exports)) of
```

Comment:
How frequently is this called?  We used to have noticeable performance issues due to similar code.  Now we prefer to use something like

```
is_exported(M, F, A) ->
    %% make a cheap test first to avoid unnecessary calls to code server
    case erlang:module_loaded(M) of
        false -> code:ensure_loaded(M);
        true -> ok
    end,
    erlang:function_exported(M, F, A).
```

Reply:
Thank you! Now I remembered.
I had a bad feeling about this function but couldn't figure out why when writing it.

Nonetheless, this is not frequent at all.
Also, this becomes a high cost only if the module does not exist?
In our case, the module is ensured to be loaded already, otherwise it can never get this far.

Reply:
If you know the module is loaded, I'd suggest to simply use erlang:function_exported/3.  It's cleaner than inspecting the exports list, and allocates less garbage.